zaza/openstack/charm_tests/cinder_netapp/tests.py

```diff
@@ -55,23 +65,40 @@ def test_cinder_config(self):
             expected_contents,
             timeout=2)
 
+    def _create_volumes(self, n):
+        """Create N volumes."""
+        volumes = []
+        for _ in range(n):
+            name = "zaza{}".format(uuid.uuid1().fields[0])
+            vol = self.cinder_client.volumes.create(
+                name=name,
+                size='1')
+            vol.reset_state('available')
+            volumes.append(vol)
+        self.zaza_volumes.extend(volumes)
+        return volumes
+
+    def _volume_host(self, vol):
+        ret = getattr(vol, 'os-vol-host-attr:host', None)
+        if ret is not None:
+            return ret
+        # A bit more roundabout, but also works when the attribute
+        # hasn't been set yet, for whatever reason.
+        for service in vol.manager.api.services.list():
+            if 'cinder-volume' in service.binary:
+                return service.host
+
     def test_create_volume(self):
-        """Test creating a volume with basic configuration."""
-        test_vol_name = "zaza{}".format(uuid.uuid1().fields[0])
-        vol_new = self.cinder_client.volumes.create(
-            name=test_vol_name,
-            size='1')
-        try:
+        """Test creating volumes with basic configuration."""
+        for vol in self._create_volumes(2):
+            self.assertIsNotNone(vol)
             openstack_utils.resource_reaches_status(
                 self.cinder_client.volumes,
-                vol_new.id,
+                vol.id,
                 wait_iteration_max_time=12000,
                 stop_after_attempt=5,
                 expected_status='available',
                 msg='Volume status wait')
-            test_vol = self.cinder_client.volumes.find(name=test_vol_name)
-            self.assertEqual(
-                getattr(test_vol, 'os-vol-host-attr:host').split('#')[0],
-                'cinder@cinder-netapp')
-        finally:
-            self.cinder_client.volumes.delete(vol_new)
+            test_vol = self.cinder_client.volumes.find(name=vol.name)
+            host = self._volume_host(test_vol)
+            self.assertIn('cinder-netapp', host)
```

Comment:
Failed with `AssertionError: 'cinder-netapp' not found in 'juju-57c895-zaza-b42470362b3b-2@LVM'` in https://review.opendev.org/c/openstack/charm-cinder-netapp/+/815130

Reply:
> Blocked by https://review.opendev.org/c/openstack/cinder/+/815560

This patch has landed in jammy/yoga for ubuntu as well as proposed pockets for impish/xena and wallaby.